### DISCUSSION TITLE
Attempting to upgrade find-any-file using bump-cask-pr results in error

### DISCUSSION THREAD

**Discussion by jcraig999** [Homebrew/discussions#2702] [Casks]
**Issue**
Attending to upgrade cask find-any-file to version 2.3.2  using `bump-cask-pr` results in 
 'fatal: A branch named 'bump-find-any-file-2.3.2' already exists.' . I have never submitted a pull request before so I had no clue what I was doing.
There doesn't appear to be an open pull request for it. So I'm not sure what this error means.

**Steps taken**

- Ran bump-cask-pr` it told me to generate a token and update that token in my .zshrc file
- did that. Ran bump-cask-pr` again. Needed to update 'git config'  with my credentials
- did that, and ran bump-cask-pr`  for a third time and then got the error below in my terminal output. 

**Terminal output**
```
josh@joshs-MBP ~ % /opt/homebrew/bin/brew bump-cask-pr --version 2.3.2 find-any-file
==> Downloading https://s3.amazonaws.com/files.tempel.org/FindAnyFile_2.3.2.zip
Already downloaded: /Users/josh/Library/Caches/Homebrew/downloads/5477cc68b5def69b6b601939d5c804d8a2e165d45eb1fde68e891b8a3e53312c--FindAnyFile_2.3.2.zip
==> replace /version\s+["']2\.3\.2["']/m with "version \"2.3.2\""
==> replace /sha256\s+["']7fc620cf8cbdd58961eb8ef37ad108d69f44b18932ac6b12817faa
audit for find-any-file: passed

1 file inspected, no offenses detected
fatal: A branch named 'bump-find-any-file-2.3.2' already exists.
Error: Failure while executing; `git checkout --no-track -b bump-find-any-file-2.3.2 origin/master` exited with 128.
josh@joshs-MBP ~ % 
```

**Comment by neutric** [comment]
Changes should not be committed directly to the master branch but in feature branches. `brew bump-cask-pr` creates those for you and names them based on cask token and version. In your case the branch is called "bump-find-any-file-2.3.2".

During one of the two unsuccessful runs that you said it took you to get set up, this branch was created as indicated by `fatal: A branch named 'bump-find-any-file-2.3.2' already exists.`

Delete it with `git branch -D bump-find-any-file-2.3.2`. After that, try `brew bump-cask-pr --version 2.3.2 find-any-file` again.

Feel free to ask further questions. If satisfied, please mark this as answered.

**Reply by jcraig999** [reply]
> Delete it with `git branch -D bump-find-any-file-2.3.2`. After that, try `brew bump-cask-pr --version 2.3.2 find-any-file` again.
> 
> Feel free to ask further questions. If satisfied, please mark this as answered.

I tried to type that command and got the following error: `fatal: not a git repository (or any of the parent directories): .git`
Then, I did `git branch -a` to list all of the branches, and I got the same error.

**Reply by carlocab** [reply]
You need to do this inside the homebrew-cask repository. Try
```sh
git -C "$(brew --repository homebrew/cask)" branch -D bump-find-any-file-2.3.2
```

**Reply by gromgit** [reply]
You need to either be in the `homebrew-cask` tap, or reference it explicitly in your `git` commands (`brew` command automatically Do The Right `git` Thing):
```
git -C $(brew --repo homebrew/cask) branch -D bump-find-any-file-2.3.2
git -C $(brew --repo homebrew/cask) branch -a
brew bump-cask-pr --version 2.3.2 find-any-file
```

**Reply by jcraig999** [reply]
> git -C $(brew --repo homebrew/cask) branch -D bump-find-any-file-2.3.2

Tried that, got the following: `git -C $(/opt/homebrew/bin/brew --repo homebrew/cask) branch -D bump-find-any-file-2.3.2
error: Cannot delete branch 'bump-find-any-file-2.3.2' checked out at '/opt/homebrew/Library/Taps/homebrew/homebrew-cask'

**Reply by carlocab** [reply]
Do `git -C "$(brew --repository homebrew/cask)" checkout master` first before any of the other `git` commands.

**Reply by jcraig999** [reply]
> Do `git -C "$(brew --repository homebrew/cask)" checkout master` first before any of the other `git` commands.

Thanks. I was able to successfully delete the branch. I will try the other commands now.

**Comment by jcraig999** [comment]


**Reply by carlocab** [reply]
Please [don't use screenshots](https://meta.stackoverflow.com/a/285557) when sharing the output of commands in your terminal. They're often difficult to read and impossible to reuse (e.g. when looking up error messages). Instead, copy the text here and wrap them in three back-ticks (```) on the line before and on the line after to format them properly.

**Comment by jcraig999** [comment]
I was able to successfully delete the branch but here is what happened when I tried to run `bump-cask-pr` . I apologize. I'm obviously not doing something correctly. Appreciate the help.
```
josh@joshs-MBP ~ % brew bump-cask-pr --version 2.3.2 find-any-file
==> Auto-updated Homebrew!
Updated Homebrew from 0b14b6d76 to 60afbd761.
Updated 2 taps (homebrew/core and homebrew/cask).
==> New Formulae
nerdctl
==> Updated Formulae
Updated 55 formulae.
==> Renamed Formulae
mr2 -> zoro
==> New Casks
gaphor                     gridtracker                qth
==> Updated Casks
Updated 27 casks.
==> Deleted Casks
jalbum

==> Downloading https://s3.amazonaws.com/files.tempel.org/FindAnyFile_2.3.2.zip
Already downloaded: /Users/josh/Library/Caches/Homebrew/downloads/5477cc68b5def69b6b601939d5c804d8a2e165d45eb1fde68e891b8a3e53312c--FindAnyFile_2.3.2.zip
==> replace /version\s+["']2\.3\.2["']/m with "version \"2.3.2\""
==> replace /sha256\s+["']7fc620cf8cbdd58961eb8ef37ad108d69f44b18932ac6b12817faa
audit for find-any-file: passed

1 file inspected, no offenses detected
Switched to a new branch 'bump-find-any-file-2.3.2'
On branch bump-find-any-file-2.3.2
nothing to commit, working tree clean
Error: Failure while executing; `git commit --no-edit --verbose --message=Update\ find-any-file\ from\ 2.3.2\ to\ 2.3.2 -- /opt/homebrew/Library/Taps/homebrew/homebrew-cask/Casks/find-any-file.rb` exited with 1.
```

**Reply by neutric** [reply]
That would be because somebody else bumped `find-any-file` to version 2.3.2. The corresponding [Pull Request](https://github.com/Homebrew/homebrew-cask/pull/116697) was already merged into the master branch and thus made it to your local version of the git repository after the Homebrew auto-update at the beginning of your log. Your run of `bump-cask-pr` computed the same `sha256` checksum, so there were no changes that could have been committed.
